### PR TITLE
Feat/FCC-23-comment-route

### PR DESCRIPTION
**Pull request by Sameer55tech**

### CODE REVIEW

File: src/core/core/comment.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+const update = async (body, id) => {
+  return await commentModel.update(body, { where: { id: id } });
+};
+
+const findOne = async (id) => {
+  return await commentModel.findByPk(id);
+};
+
+const findMany = async (postId) => {
+  return await commentModel.findAll({
```
Comment (luckytarun24) [inline src/core/core/comment.js]: make it dynamic just take the query as props.
like -  comment.findMany({id:id})

const findMany = async (query) => {
  return await commentModel.findAll(query)

File: src/core/core/comment.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    ],
+  });
+};
+
+const remove = async (id) => {
+  return await commentModel.destroy({ where: { id: id } });
+};
+
+const removeMany = async (postId) => {
+  return await commentModel.destroy({ where: { postId: postId } });
```
Comment (luckytarun24) [inline src/core/core/comment.js]: same as findmany

File: src/core/models/comment/commentModel.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    },
+  },
+  {
+    timestamps: true,
+    freezeTableName: true,
+  }
+);
+
+comment.belongsTo(user, { foreignKey: "userId" });
+// comment.belongsTo(post, { foreignKey: "postId" });
```
Comment (luckytarun24) [inline src/core/models/comment/commentModel.js]: remove comment

File: src/core/routes/authorized/comment.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import { commentController } from "../../controller";
+
+const router = express.Router();
+
+router.post("/", commentController.createComment);
+router.patch("/update/:id", commentController.updateComment);
+router.delete("/delete/:id", commentController.deleteOneComment);
+router.get("/:postId", commentController.getComments);
+router.get("/:id", commentController.getOneComment);
+router.delete("/delete/:postId", commentController.deleteComments);
```
Comment (luckytarun24) [inline src/core/routes/authorized/comment.js]: line 7 and 11 both routes are differentiate them

File: src/core/routes/authorized/comment.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import express from "express";
+import { commentController } from "../../controller";
+
+const router = express.Router();
+
+router.post("/", commentController.createComment);
+router.patch("/update/:id", commentController.updateComment);
+router.delete("/delete/:id", commentController.deleteOneComment);
+router.get("/:postId", commentController.getComments);
```
Comment (luckytarun24) [inline src/core/routes/authorized/comment.js]: line 9 and 10 both are same differentiate them

**Review by tarunjain55 — APPROVED** [review]

**Pull request merged**